### PR TITLE
Add `yield` operator

### PR DESCRIPTION
**Pull request by jachris**
The remaining todos are mostly something for the language revamp. Also, the operator is intentionally a bit under-specified because we are not sure yet how exactly nulls, unknown fields, etc., will be handled eventually.

### CODE REVIEW

File: web/docs/operators/transformations/yield.md [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+## Examples
+
+The schema `suricata.dns` provides a list of answers for DNS queries. We want
+to extract all answers for `CNAME` records.
+
+```
+from eve.json |
+where #schema == "suricata.dns" |
+yield dns.answers[] |
+where rrtype == "CNAME"
```
Comment (mavam) [inline web/docs/operators/transformations/yield.md:30]:
```suggestion
from eve.json
| where #schema == "suricata.dns"
| yield dns.answers[]
| where rrtype == "CNAME"
```

I know why you put the pipe last, but this is consistent with the rest of the examples. 😉

File: libtenzir/builtins/operators/yield.cpp [hunk 2]
```diff
@@ -0,0 +1,217 @@
...
+public:
+  yield_operator() = default;
+
+  explicit yield_operator(parser_interface& p) {
+    auto parser = argument_parser{"yield", "https://docs.tenzir.com/next/"
+                                           "operators/transformations/yield"};
+    auto extractor = located<std::string>{};
+    // TODO: This assumes that the given extractor can be parsed as a shell-like
+    // argument, e.g., spaces must be quoted.
+    parser.add(extractor, "<extractor>");
```
Comment (dominiklohmann) [inline libtenzir/builtins/operators/yield.cpp:40]: I think we need to discuss whether we want `.'foo bar'.baz` or `.["foo bar"].baz`. Both cases are also a strong argument for requiring fields to start with a `.` to make parsing easier.

I think we can postpone this and merge the operator regardless, though.

**Review by dominiklohmann — APPROVED** [review]
This is great, thank you.

File: libtenzir/builtins/operators/yield.cpp (hunk 2 above)
Comment (jachris) [reply libtenzir/builtins/operators/yield.cpp:40]: Agreed. Let's make this part of the language revamp.

**Pull request merged**